src/core/audiosource.cpp

```diff
@@ -215,10 +215,29 @@ void FFMS_AudioSource::SetOutputFormat(FFMS_ResampleOptions const& opt) {
 			"Could not open avresample context");
 	newContext.swap(ResampleContext);
 #endif
+#ifdef WITH_SWRESAMPLE
```

Comment:
There's a lot of unnecessary duplication here, and this'll do broken things if both avresample and swresample are enabled.